### PR TITLE
pdftk-java: new port

### PR DESCRIPTION
**Pull request by chrstphrchvz**
(Depends on PR #4925)

Based on submission by @someuser12: https://trac.macports.org/ticket/58757

A few adjustments from the submitted portfile:
 - Use `GPL-2+` as license, since source files state "version 2 of the License, or (at your option) any later version"
 - Use java portgroup: pdftk-java requires Java 7 or higher. Use latest LTS Java (OpenJDK 11) as fallback.
 - Use `3.0.6` as version instead of commit date; I think that would be friendlier, especially for a non-`devel` port. I have added a brief comment explaining the use of the non-release commit (it is the commit immediately following 3.0.6 for a minor bugfix; I think using the commit is more reasonable than using the 3.0.6 release and then adding a patch for that fix).
 - Specify dependencies with `-append`; move `bcprov` and `commons-lang3` to library dependencies (since these are both build and runtime dependencies).

###### ~~Tested on~~

~~macOS 10.x~~
~~Xcode 8.x~~

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by chrstphrchvz** [comment]
```
Error: Failed to destroot pdftk-java: xinstall: Cannot stat: /Users/travis/build/macports/macports-ports/textproc/pdftk-java/files/pdftk-java, No such file or directory
```

I don't know what the cause of this is error.

@someuser12, were you able to successfully install this port?

**Comment by someuser12** [comment]
@chrstphrchvz, thanks for your work on these three port submissions.

I apologize, I had forgotten to join to the [original trac ticket](https://trac.macports.org/ticket/58757) the pdftk-java script which sould go into the files subdirectory of the port and which is referenced by the portfile. I just joined it to the trac ticket. This was the reason of the build failure (for me, the build was successful under Mac OS 10.11.6 with XCode 7.2).

File: textproc/pdftk-java/Portfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 8; truncate-lines: t; indent-tabs-mode: nil; c-basic-offset: 4
+# vim: set fileencoding=utf-8 tabstop=8 shiftwidth=4 softtabstop=4 noexpandtab filetype=tcl :
+
+PortSystem 1.0
+PortGroup            java 1.0
+
+#PortGroup           github 1.0
```
[the remark this answers is not in the document]
Comment (chrstphrchvz) [reply textproc/pdftk-java/Portfile]: Removed in 98500f8956

File: textproc/pdftk-java/Portfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                    port:commons-lang3
+
+worksrcdir          ${git_project}-${git_commit}
+
+use_configure       no
+
+pre-build {
+    file mkdir ${worksrcpath}/lib
+    system "cd ${worksrcpath}/lib && ln -s ${prefix}/share/java/bcprov/bcprov.jar"
+    system "cd ${worksrcpath}/lib && ln -s ${prefix}/share/java/commons-lang3.jar"
```
[the remark this answers is not in the document]
Comment (chrstphrchvz) [reply textproc/pdftk-java/Portfile]: Thanks, resolved in 98500f8956

File: textproc/pdftk-java/Portfile [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+master_sites        https://gitlab.com/pdftk-java/${git_project}/-/archive/${git_commit}/
+
+checksums           rmd160  02546b6b1553348f4b5b74fd421a8ed2a84bd282 \
+                    sha256  306c73693bc24fa7b6ea32967a7028cb027b4658f4eae96cb3a046c413da07d8 \
+                    size    1360035
+
+java.version        1.7+
+java.fallback       openjdk11
+
+depends_build-append bin:ant:apache-ant
```
[the remark this answers is not in the document]
Comment (chrstphrchvz) [reply textproc/pdftk-java/Portfile]: Should be resolved in 98500f8956

**Pull request merged**